cmd/integration_test.go

```diff
@@ -389,3 +389,50 @@ func TestSubMetricThresholdNoData(t *testing.T) {
        { tag:xyz }........: 0   0/s
      two..................: 42`)
 }
+
+func TestSetupTeardownThresholds(t *testing.T) {
+	t.Parallel()
+	tb := httpmultibin.NewHTTPMultiBin(t)
+
+	script := []byte(tb.Replacer.Replace(`
+		import http from "k6/http";
+		import { check } from "k6";
+		import { Counter } from "k6/metrics";
+
+		let statusCheck = { "status is 200": (r) => r.status === 200 }
+		let myCounter = new Counter("setup_teardown");
+
+		export let options = {
+			iterations: 5,
```

Comment:
I guess we have a test that "fails" in the same way this is checking it won't fail?

Reply:
Sorry, I don't understand what you mean

Reply:
This test does check that if we do something it *works*. We sometimes (probably should do it more) write the reverse test where we check that the reverse of what should work ... does not work. 

having read the test in more detail now ... I am not certain what that will be in this case though as we seem to not be checking that the threshold parsing works (which is what I though it did reading your commit message). But instead it seems to be about teardown and setup emitting metrics that can be used in thresholds.

So I am not certain how we can check that this test doesn't just pass as we do not actually check anything in it.  

I am also less certain that moving this to `cmd/` makes a difference :shrug:

Reply:
I see. The integration tests automatically check if the exit code is 0 by default, so if the thresholds had failed, it would catch it. If I change the `options.thresholds`, the test fails, when before it didn't.

Notice also the checkmarks in the end-of-test summary checks at the end, e.g. `✓ http_reqs......................: 7`. That's another guarantee that the thresholds were actually correctly parsed and evaluated that I added to the test.

Reply:
hmm though you are right, we don't have an integration test that checks if some thresholds have failed :thinking: I'll make a simple one in a bit

Reply:
resolved by https://github.com/grafana/k6/pull/2625/commits/ad33cbbc95e3f61af5c1a6b971156883c4ab6939